automerge-backend/src/backend.rs

```diff
@@ -26,6 +30,8 @@ pub struct Backend {
     actors: ActorMap,
     history: Vec<Change>,
     history_index: HashMap<amp::ChangeHash, usize>,
+    /// A cache of vector clocks for speeding up sync operations.
+    clocks_cache: Arc<Mutex<HashMap<amp::ChangeHash, VectorClock>>>,
```

Comment:
@alexjg Currently this map doesn't shrink, I'm not sure of a good way to decide when to do that besides a fixed size cache, but then what size do we use? Any thoughts?